subsys/emul/espi/espi_emul_host.c

```diff
@@ -0,0 +1,388 @@
+
+	if (idx < 0 || data->vw_state[idx].dir != ESPI_MASTER_TO_SLAVE) {
+		return -EPERM;
+	}
+
+	*level = data->vw_state[idx].level;
+
+	evt.evt_type = ESPI_BUS_EVENT_VWIRE_RECEIVED;
+	evt.evt_details = vw;
+	evt.evt_data = *level;
+
+	espi_emul_raise_event(data->espi, evt);
+
+	return 0;
+}
+
+static int host_espi_api_internal_set_vwire(const struct espi_emul *dev,
+				   enum espi_vwire_signal vw,
+				   uint8_t level)
+{
+	struct espi_host_emul_data *data =
+		CONTAINER_OF(dev, struct espi_host_emul_data, emul);
+	struct espi_event evt;
+	int idx;
+
+	idx = emul_host_find_index(data, vw);
+
+	if (idx < 0 || data->vw_state[idx].dir != ESPI_MASTER_TO_SLAVE) {
+		return -EPERM;
+	}
+
+	data->vw_state[idx].level = level;
+
+	evt.evt_type = ESPI_BUS_EVENT_VWIRE_RECEIVED;
+	evt.evt_details = vw;
+	evt.evt_data = level;
+
+	espi_emul_raise_event(data->espi, evt);
+
+	return 0;
+}
+
+/* Logical Channel 2 APIs */
+static int host_espi_api_send_oob(const struct espi_emul *dev,
+				 struct espi_oob_packet *pckt)
+{
+	return -EIO;
+}
+
+static int host_espi_api_receive_oob(const struct espi_emul *dev,
+				    struct espi_oob_packet *pckt)
+{
+	return -EIO;
+}
+
+/* Logical Channel 3 APIs */
+static int host_espi_api_flash_read(const struct espi_emul *dev,
+				   struct espi_flash_packet *pckt)
+{
+	return -EIO;
+}
+
+static int host_espi_api_flash_write(const struct espi_emul *dev,
+				    struct espi_flash_packet *pckt)
+{
+	return -EIO;
+}
+
+static int host_espi_api_flash_erase(const struct espi_emul *dev,
+				    struct espi_flash_packet *pckt)
+{
+	return -EIO;
+}
+
+/* Callbacks and traffic intercept */
+static int host_espi_api_manage_callback(const struct espi_emul *dev,
+					struct espi_callback *callback,
+					bool set)
+{
+	struct espi_host_emul_data *data =
+		CONTAINER_OF(dev, struct espi_host_emul_data, emul);
+
+	return espi_manage_callback(&data->callbacks, callback, set);
+}
+
+int host_espi_api_raise_event(const struct espi_emul *dev, struct espi_event ev)
+{
+	struct espi_host_emul_data *data =
+		CONTAINER_OF(dev, struct espi_host_emul_data, emul);
+
+	espi_send_callbacks(&data->callbacks, data->espi, ev);
+
+	return 0;
+}
+
+static struct espi_emul_device_api espi_emul_host_api = {
+	.config = host_espi_api_config,
+	.get_channel_status = host_espi_api_get_channel_status,
+	.read_request = host_espi_api_read_request,
+	.write_request = host_espi_api_write_request,
+	.read_lpc_request = host_espi_api_lpc_read_request,
+	.write_lpc_request = host_espi_api_lpc_write_request,
+	.send_vwire = host_espi_api_send_vwire,
+	.receive_vwire = host_espi_api_receive_vwire,
+	.internal_set_vwire = host_espi_api_internal_set_vwire,
+	.send_oob = host_espi_api_send_oob,
+	.receive_oob = host_espi_api_receive_oob,
+	.flash_read = host_espi_api_flash_read,
+	.flash_write = host_espi_api_flash_write,
+	.flash_erase = host_espi_api_flash_erase,
+	.manage_callback = host_espi_api_manage_callback,
+	.raise_event = host_espi_api_raise_event,
+};
```

Comment:
Some of there don't have `static`, is that on purpose?

Reply:
No, fixed